pkg/resource/endpoint/custom_update_conditions.go

```diff
@@ -17,48 +17,26 @@
 package endpoint
 
 import (
-	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
 	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
-	"github.com/aws/aws-sdk-go/aws"
+	customShared "github.com/aws-controllers-k8s/sagemaker-controller/pkg/common"
 	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
-	corev1 "k8s.io/api/core/v1"
 )
 
+var terminalMessage = "endpoint status: Failed. Check FailureReason"
+
 // CustomUpdateConditions sets conditions (terminal) on supplied endpoint
 // it examines supplied resource to determine conditions.
 // It returns true if conditions are updated
-func (rm *resourceManager) customUpdateConditions(
+func (rm *resourceManager) CustomUpdateConditions(
 	ko *svcapitypes.Endpoint,
 	r *resource,
 	err error,
 ) bool {
 	latestStatus := r.ko.Status.EndpointStatus
-
-	if latestStatus == nil || *latestStatus != svcsdk.EndpointStatusFailed {
-		return false
-	}
-	var terminalCondition *ackv1alpha1.Condition = nil
-
-	for _, condition := range ko.Status.Conditions {
-		if condition.Type == ackv1alpha1.ConditionTypeTerminal {
-			terminalCondition = condition
-			break
-		}
-	}
-	if terminalCondition != nil && terminalCondition.Status == corev1.ConditionTrue {
-		// some other exception already put the resource in terminal condition
-		return false
-	}
-
-	// setting terminal condition since controller can no longer recover by retrying
-	if terminalCondition == nil {
-		terminalCondition = &ackv1alpha1.Condition{
-			Type: ackv1alpha1.ConditionTypeTerminal,
-		}
-		ko.Status.Conditions = append(ko.Status.Conditions, terminalCondition)
-	}
-	terminalCondition.Status = corev1.ConditionTrue
-	terminalCondition.Message = aws.String("endpoint status: Failed. check FailureReason")
-
-	return true
+	terminalStatus := svcsdk.EndpointStatusFailed
+	koStatusConditions := &ko.Status.Conditions
+	terminalMessageAddress := &terminalMessage
```

Comment:
we don't need a new variable here  
you can just pass `&terminalMessage` to the function

Reply:
Can we avoid the global string entirely ?